File: spark-core/src/main/java/com/atguigu/lh/Test01/Test01.java
package com.atguigu.lh.Test01;

import java.util.Random;

public class Test01
{
    public static void main(String[] args) {
        int[] S = new int[10000];
        int N = S.length;
        Random random = new Random();
        //生成一万个数的数组
        for (int r = 0;r < N; r ++){
            S[r] = random.nextInt(10000);
        }

        int k = 10;
        int[] R = new int[k];
        //S前K个数填充R数组
        for (int f = 0;f < k; f++){
            R[f] = S[f];
        }
        int j ;
        //遍历数组S,根据算法,替换R数组中的元素,最终生成结果R数组.
        for (int i = k;i < S.length;i++){
            j = random.nextInt(i);
            if (j < k)  R[j] = S[i];
        }
        //打印R数组的结果
        for (int i =0;i < R.length;i++) {
            System.out.println(R[i]);
        }
    }
}

File: spark-core/src/main/scala/com/atguigu/spark/core/day04/Test01.scala
package com.atguigu.spark.core.day04

import org.apache.spark.rdd.RDD
import org.apache.spark.util.LongAccumulator
import org.apache.spark.{SparkConf, SparkContext}

object Test01 {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("Practice").setMaster("local[2]")
    val sc = new SparkContext(conf)
    val rdd2: RDD[Int] = sc.parallelize(Array(3, 5, 8, 2))
    var a=1
    val rdd3: RDD[(Int, Int)] = rdd2.map(x => {
      a += 1
      println(a)
      (x, 1)
    })

    rdd3.collect()
    println("----------------")
    println(a)
    sc.stop()

  }
}
